transform-plugins/src/main/java/co/cask/hydrator/plugin/Run.java

```diff
@@ -0,0 +1,235 @@
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import com.google.common.util.concurrent.SettableFuture;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Run Plugin - Runs executable binary installed and available on the Hadoop nodes.
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name("Run")
+@Description("This plugin allows user to run an executable binary installed and available on the local filesystem of" +
+  " the Hadoop nodes. Plugin allows the user to read the structured record as input and returns the output record, " +
+  "to be further processed downstream in the pipeline.")
+public class Run extends Transform<StructuredRecord, StructuredRecord> {
+  private final RunConfig config;
+  private Schema outputSchema;
+  private RunExternalProgramExecutor executor;
+
+  public Run(RunConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
+    config.validateBinaryExecutableType();
+    config.validateInputFields(inputSchema);
+    config.validateOutputFieldType();
+    stageConfigurer.setOutputSchema(buildOutputSchema(inputSchema));
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    executor = new RunExternalProgramExecutor(config.commandToExecute);
+    executor.startAndWait();
+  }
+
+  @Override
+  public void transform(StructuredRecord structuredRecord, Emitter<StructuredRecord> emitter) throws Exception {
+    outputSchema = outputSchema == null ? buildOutputSchema(structuredRecord.getSchema()) : outputSchema;
+    StringBuilder executableCommand = new StringBuilder();
+
+    if (!Strings.isNullOrEmpty(config.fieldsToProcess)) {
+      for (String inputField : Splitter.on(',').trimResults().split(config.fieldsToProcess)) {
+        Schema inputFieldSchema = structuredRecord.getSchema().getField(inputField).getSchema();
+        Schema.Type inputFieldType = inputFieldSchema.isNullable() ? inputFieldSchema.getNonNullable().getType() :
+          inputFieldSchema.getType();
+
+        if ((inputFieldType.equals(Schema.Type.STRING)) &&
+          ((String) (structuredRecord.get(inputField))).contains(" ")) {
+          executableCommand.append("\"");
+          executableCommand.append(structuredRecord.get(inputField));
+          executableCommand.append("\"");
+        } else {
+          executableCommand.append(structuredRecord.get(inputField));
+        }
+        executableCommand.append(" ");
```

Comment:
So the contract to the external program is always a line with space separated fields? Can it be configurable? This could affect the logic to determine if quoting is needed.